packages/api-derive/src/staking/info.ts

```diff
@@ -2,42 +2,111 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import { DerivedStaking } from '../types';
+import { ApiInterface$Rx } from '@polkadot/api/types';
+import { AccountId, BlockNumber, Exposure, Option, StakingLedger,StructAny, ValidatorPrefs, UnlockChunk } from '@polkadot/types';
+import { DerivedStaking, DerivedUnlocking } from '../types';
 
-import { Observable, of } from 'rxjs';
+import BN from 'bn.js';
+import { combineLatest, Observable, of } from 'rxjs';
 import { map, switchMap } from 'rxjs/operators';
-import { ApiInterface$Rx } from '@polkadot/api/types';
-import { AccountId, Exposure, Option, StakingLedger, StructAny, ValidatorPrefs } from '@polkadot/types';
 
+import { bestNumber } from '../chain/bestNumber';
 import { drr } from '../util/drr';
+import { eraLength } from '../session/eraLength';
+import { isUndefined } from '@polkadot/util';
+
+function calculateUnlocking (stakingLedger: StakingLedger | undefined, eraLength: BN, bestNumber: BlockNumber): DerivedUnlocking | undefined {
+  if (isUndefined(stakingLedger)) {
+    return undefined;
+  }
+
+  // select the Unlockchunks that can't be redeemed yet.
+  const unlockingChunks = stakingLedger.unlocking.filter((chunk) => remainingBlocks(chunk.era, eraLength, bestNumber).gtn(0));
+
+  if (!unlockingChunks.length) {
+    return undefined;
+  }
+
+  // group the Unlockchunks that have the same era and sum their values
+  const groupedResult = groupByEra(unlockingChunks);
+  const results: DerivedUnlocking = [];
+
+  Object.keys(groupedResult).map((eraString) => (
```

Comment:
As discussed, this would probably be a better form -

```js
const results = Object.entries(groupedResult).map(([eraString, value]) => ({
    value,
    remainingBlocks: remainingBlocks(new BlockNumber(eraString), eraLength, bestNumber)
}));
```

Reply:
Thanks. I haven't pushed the modifications yet as I make my way on an e2e test.